testing/iframe.js

```diff
@@ -319,25 +320,42 @@ export function createIframeWithMessageStub(win) {
    * Returns a Promise that resolves when the iframe acknowledged the reception
    * of the specified message.
    */
-  element.expectMessageFromParent = msg => {
-    return new Promise(resolve => {
-      const listener = event => {
-        let expectMsg = msg;
-        let actualMsg = event.data.receivedMessage;
-        if (typeof expectMsg !== 'string') {
-          expectMsg = JSON.stringify(expectMsg);
-          actualMsg = JSON.stringify(actualMsg);
+  element.expectMessageFromParent = (type, callback) => {
+    if (typeof type === 'function') {
```

Comment:
Yeah, I think it's not very straightforward for people to use in tests. For example I would have to go through the code to figure out what a callback function should look like.
Maybe update the comment for the function description? At least we should document it clear.